pkg/bpf/map_linux.go

```diff
@@ -986,6 +988,18 @@ func (m *Map) Delete(key MapKey) error {
 	return err
 }
 
+// Delete deletes the map entry corresponding to the given key.
+// Error metrics are not incremented for missing entries, however
+// an error is still returned for a missing entry.
+func (m *Map) DeleteIgnoringMissing(key MapKey) error {
```

Comment:
A bit iffy. Better API to return (bool, error) to indicate whether it was deleted or not? Downstream code in ctmap.go relies on error to decide whether to proceed with deleting nat mapping.

Reply:
Alternatively could call this SilentDelete() as suggested here https://github.com/cilium/cilium/pull/11526#discussion_r430816094.

Reply:
Went with SilentDelete().

Reply:
Good catch. Changed the API to return `(bool, err)`.